libraries/abstractions/posix/include/FreeRTOS_POSIX/pthread.h

```diff
@@ -243,7 +243,7 @@ int pthread_barrier_wait( pthread_barrier_t * barrier );
  */
 int pthread_create( pthread_t * thread,
                     const pthread_attr_t * attr,
-                    void *( *startroutine )( void * ),
+                    void * ( *startroutine )( void * ),
```

Comment:
Did this change come out of uncrustify? I do not like changing the pthread code in a PR about units tests, do we really need this change in this PR?

Reply:
This change is to simulate better the environment, since the secure_sockets code is calling xTaskCreate to create a thread to notify the library user of changes on the descriptor.
I tried to make it simple before, instead of launching a thread, i just call the function, but i was impossible to control without changing the code itself, and the address sanitizers complained about usage after free.

Reply:
i removed the dependency over pthread by creating a library, and will have to be implemented per supported platform, current implementation is for linux with pthread